bluemira/equilibria/find.py

```diff
@@ -725,6 +728,90 @@ def get_flux_loop(psi_norm):
     return lcfs, separatrix
 
 
+def get_legs(equilibrium):
+    """
+    Get the legs of a separatrix.
+
+    Parameters
+    ----------
+    equilibrium: Equilibrium
+        Equilibrium for which to find the separatrix legs
+
+    Returns
+    -------
+    legs: List[Loop]
+        Geometries of the legs, sorted as:
+            [lower inner,  # if present
+             lower outer,  # if present
+             upper inner,  # if present
+             upper outer,  # if present
+            ]
+
+    Raises
+    ------
+    EquilibriaError: if a strange number of legs would be found for an X-point
+
+    Notes
+    -----
+    Will return two legs in the case of a single null
+    Will return four legs in the case of a double null
+    """
+
+    def extract_leg(flux_line, x_point):
+        radial_line = Loop(
+            x=[x_point.x - delta, x_point.x + delta], z=[x_point.z, x_point.z]
+        )
+        arg_inters = join_intersect(flux_line, radial_line, get_arg=True)
+        arg_inters.sort()
+        if x_point.z < o_p.z:
+            # Lower null
+            func = operator.lt
+        else:
+            # Upper null
+            func = operator.gt
+
+        if len(arg_inters) > 2:
+            EquilibriaError(
+                "Unexpected number of intersections with the separatrix around the X-point."
+            )
+
+        flux_legs = []
+        for arg in arg_inters:
+            if func(flux_line.z[arg + 1], flux_line.z[arg]):
+                leg = Loop.from_array(flux_line[arg:])
+            else:
+                leg = Loop.from_array(flux_line[: arg + 1])
+
+            # Make the leg flow away from the plasma core
+            if leg.argmin(x_point[:2]) > 3:
+                leg.reverse()
+
+            flux_legs.append(leg)
+        if len(flux_legs) == 1:
+            flux_legs = flux_legs[0]
+        return flux_legs
+
+    o_points, x_points = equilibrium.get_OX_points()
+    o_p = o_points[0]
+    separatrix = equilibrium.get_separatrix()
+    delta = 2 * equilibrium.grid.dx
+
+    if isinstance(separatrix, Iterable):
+        # Double null
+        legs = []
+        for half_sep in separatrix:
+            for x_p in x_points[:2]:
+                leg = extract_leg(half_sep, x_p)
+                legs.append(leg)
+    else:
+        # Single null
+        legs = extract_leg(separatrix, x_points[0])
+
+    # Sort [lower inner, lower outer, upper inner, upper outer]
+    legs.sort(key=lambda leg: leg.x[0] + leg.z[0])
+    return legs
```

Comment:
How would you feel about using a `dict` instead of a list as a return type here? Then, when we're using the return value, it's immediately clear what leg we're accessing, e.g.,

```python
legs = get_legs(eq)
Divetor.make_target(legs["LOWER_INNER"])
```

As we have a known set of keys, we could also consider using an `Enum`. Then the typing tells us exactly what keys are allowed:

```python
class Leg(Enum):
    LOWER_INNER = "lower_inner"
    LOWER_OUTER = "lower_outer"
    ...

def get_legs(equilibrium) -> Dict[Leg, Loop]:
    ...
```

Reply:
With the enum it would be possible to keep the list and just set `LOWER_INNER = 0` etc. No preference however

Reply:
> Am I right in thinking that this outputs a single loop for each leg? The old `StreamFlow` implementation outputted several loops (scrape off layers?). Are those other layers no longer needed?
> 
> I've left a couple of comments on the code, let me know what you think

Yes, this would just get you the legs of the actual separatrix. Do you need more layers? Can add them in, I can certainly see how it might help divertor design, just wasn't sure if you needed them for what is already there.

Reply:
> With the enum it would be possible to keep the list and just set `LOWER_INNER = 0` etc. No preference however

Happy to put in a dict and enum here, yep